ecl/hthor/hthor.cpp

```diff
@@ -2402,15 +2420,20 @@ const void * CHThorGroupDedupKeepLeftActivity::nextRowGE(const void * seek, unsi
             break;
         }
 
-        if (numKept < numToKeep-1)
+        if (compareBest)
+        {
+            if (compareBest->docompare(prev,next) < 0 )
```

Comment:
Is this test condition correct?  It is checked inconsistently - sometimes <, sometimes > sometimes <=.

Does BEST(surname),BEST(score) mean the first or the last?  From your documentation it is mean to be the first, I think this will return the last.